src/Talis/Persona/Client/Base.php

```diff
@@ -290,16 +290,16 @@ protected function performRequest($url, array $opts)
                 array("opts" => $opts, "url" => $url, "response" => $response)
             );
 
-            switch ($headers['http_code']) {
+            switch ($response->getStatusCode()) {
             case 404:
                 throw new NotFoundException(
                     "Received 404 response from persona",
-                    $headers['http_code']
+                    $response->getStatusCode()
```

Comment:
It might be impossible to get to line 297. An exception is thrown for any non successful response which is dealt with on line 273.